### PR TITLE
[KYUUBI #306] SparkProcessBuilder support Windows

### LINKED ISSUE #306: Spark Engine should support windows
Easy to test for developers using Windows.

### PR DESCRIPTION
**Pull request by Honglun**
Honglun [Closes #308](https://github.com/yaooqinn/kyuubi/pull/308)    Test Plan Feature [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### Please add issue ID here?

Fixes #306

### Why are the changes needed?

SparkProcessBuilder support Windows. Easy to test for developers using Windows.

### Test Plan:
- Add some test cases that check the changes thoroughly including negative and positive cases if possible
- Add screenshots for manual tests if appropriate
- [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by ulysses-you** [comment]
thanks for the contributing @Honglun. Can you provide some screen shot or useful log in your env ?

File: kyuubi-main/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala [hunk 1]
```diff
@@ -159,4 +159,11 @@ object SparkProcessBuilder {
   private final val PROXY_USER = "--proxy-user"
   private final val PRINCIPAL = "spark.kerberos.principal"
   private final val KEYTAB = "spark.kerberos.keytab"
+  private final val SPARK_SUBMIT_FILE = {
+    if (String.valueOf(System.getProperty("os.name")).toLowerCase().contains("win")) {
```
Comment (yaooqinn) [inline kyuubi-main/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala:163]: We can add some thing like below in `kyuubi-common` `Utils` object
```scala
import org.apache.commons.lang3.SystemUtils.IS_OS_WINDOWS

...

/**
 * Whether the underlying operating system is Windows.
 */
val isWindows = SystemUtils.IS_OS_WINDOWS

```

If we want to support win, we can make such a step for further update. WDYT

**Comment by yaooqinn** [comment]
@Honglun Thanks for your first contribution and welcome! Just left some minor comments.

**Comment by Honglun** [comment]
@ulysses-you  Are the following logs OK? 
image

File: kyuubi-main/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala (hunk 1 above)
Comment (Honglun) [reply kyuubi-main/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala:163]: I agree. It's better.

Comment (yaooqinn) [reply kyuubi-main/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala:163]: can you make a change in this way?

Comment (Honglun) [reply kyuubi-main/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala:163]: OK

File: kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala [hunk 2]
```diff
@@ -24,8 +24,11 @@ import java.util.{Properties, UUID}
 
 import scala.collection.JavaConverters._
 
+import org.apache.commons.lang.SystemUtils
 import org.apache.hadoop.security.UserGroupInformation
 
+
+
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala:31]: these are unnecessary, can we remove?

Comment (Honglun) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala:31]: OK. I have no problem testing it on my env, I will test it on Linux before committing.

Comment (Honglun) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala]: Could you help delete the user information of 003006? I just found that I used the wrong user when committing.

Comment (yaooqinn) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala]: you can set the right username locally,   then  re-push this by `push -f`

File: kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala [hunk 3]
```diff
@@ -150,4 +151,9 @@ private[kyuubi] object Utils extends Logging {
           s" version string, but it could not find the major and minor version numbers.")
     }
   }
+
+  /**
+  * Whether the underlying operating system is Windows.
+    */
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala:157]: style

**Pull request closed without merging**